artemis-journal/src/main/java/org/apache/activemq/artemis/core/journal/impl/JournalImpl.java

```diff
@@ -3267,20 +3279,30 @@ void scheduleReclaim() {
          return;
       }
 
+      if (logger.isDebugEnabled()) {
+         logger.debug("JournalImpl::scheduleReclamin, autoReclaim={}, compactorRunning={}", isAutoReclaim(), compactorRunning.get());
+      }
+
       if (isAutoReclaim() && !compactorRunning.get()) {
+         logger.trace("Scheduling reclaim and compactor checks");
          compactorExecutor.execute(new Runnable() {
             @Override
             public void run() {
                try {
                   processBackup();
-                  if (!checkReclaimStatus()) {
-                     checkCompact();
-                  }
+                  checkReclaimStatus();
+                  checkCompact();
                } catch (Exception e) {
                   ActiveMQJournalLogger.LOGGER.errorSchedulingCompacting(e);
+               } finally {
+                  logger.debug("JournalImpl::scheduleReclaim finished");
                }
             }
          });
+      } else {
+         if (logger.isTraceEnabled()) {
+            logger.trace("Ignoring compactorExecutor call because of autoReclaim={} and compactorRunning={}", isAutoReclaim(), compactorRunning.get());
```

Comment:
These values on the other hand, seem like they could actually change between first inspection in the if and later being logged, when the values are retrieved again.

If getting accurate info on these is important, would need to assigning the values so the logging reflects the values used for the decision.

If not, would maybe be worth tweaking the log message (e.g dropping 'because of', add some brackets, say 'current') to make it clearer the values logged are current and actually separate from the earlier decision.

Reply:
I think it is ok